src/server/gc_worker/gc_worker.rs

```diff
@@ -727,15 +800,31 @@ where
         max_ts: TimeStamp,
         start_key: &Key,
         limit: usize,
+        regions_provider: Arc<dyn RegionInfoProvider>,
     ) -> Result<Vec<LockInfo>> {
-        let snap = self
-            .engine
-            .snapshot_on_kv_engine(start_key.as_encoded(), &[])
-            .unwrap();
-        let mut reader = MvccReader::new(snap, Some(ScanMode::Forward), false);
-        let (locks, _) = reader
-            .scan_locks(Some(start_key), None, |l| l.ts <= max_ts, limit)
-            .map_err(TxnError::from_mvcc)?;
+        let regions = box_try!(regions_provider.get_regions_in_range(start_key.as_encoded(), &[]))
```

Comment:
We need to evaluate the performance impact of changing from one seek to multiple seek.

Reply:
As Green GC is going to be deprecated, I think we don't need to concern about it too much.
Maybe it is the time to deprecate it now, since only scanning regions found in region_info_provider may possibly affect the scanning's correctness.

Reply:
Indeed.